apps/editor/src/app/routes/settingList.tsx

```diff
@@ -210,81 +178,83 @@ function SettingList() {
         t('errorMessages.invalidRange', {
-          min: peeringTimeoutMs.settingRestriction?.minValue ?? 1000,
-          max: peeringTimeoutMs.settingRestriction?.maxValue ?? 10000
+          min: settings[SettingName.PeeringTimeoutMs].settingRestriction?.minValue ?? 1000,
+          max: settings[SettingName.PeeringTimeoutMs].settingRestriction?.maxValue ?? 10000
         })
       ),
-    invoiceTries: NumberType()
+    [SettingName.InvoiceReminderMaxTries]: NumberType()
       .isRequired(t('errorMessages.required'))
       .range(
-        invoiceReminderTries.settingRestriction?.minValue ?? 0,
-        invoiceReminderTries.settingRestriction?.maxValue ?? 10,
+        settings[SettingName.InvoiceReminderMaxTries].settingRestriction?.minValue ?? 0,
+        settings[SettingName.InvoiceReminderMaxTries].settingRestriction?.maxValue ?? 10,
         t('errorMessages.invalidRange', {
-          min: invoiceReminderTries.settingRestriction?.minValue ?? 0,
-          max: invoiceReminderTries.settingRestriction?.maxValue ?? 10
+          min: settings[SettingName.InvoiceReminderMaxTries].settingRestriction?.minValue ?? 0,
+          max: settings[SettingName.InvoiceReminderMaxTries].settingRestriction?.maxValue ?? 10
         })
       ),
-    invoiceFrequency: NumberType()
+    [SettingName.InvoiceReminderFreq]: NumberType()
       .isRequired(t('errorMessages.required'))
       .range(
-        invoiceReminderFreq.settingRestriction?.minValue ?? 0,
-        invoiceReminderFreq.settingRestriction?.maxValue ?? 30,
+        settings[SettingName.InvoiceReminderFreq].settingRestriction?.minValue ?? 0,
+        settings[SettingName.InvoiceReminderFreq].settingRestriction?.maxValue ?? 30,
         t('errorMessages.invalidRange', {
-          min: invoiceReminderFreq.settingRestriction?.minValue ?? 0,
-          max: invoiceReminderFreq.settingRestriction?.maxValue ?? 30
+          min: settings[SettingName.InvoiceReminderFreq].settingRestriction?.minValue ?? 0,
+          max: settings[SettingName.InvoiceReminderFreq].settingRestriction?.maxValue ?? 30
         })
       )
   })
 
+  const formValue = Object.values(settings).reduce(
+    (values, setting) => ({
+      ...values,
+      [setting.name]: setting.value
+    }),
+    {} as Record<SettingName, unknown>
+  )
+
   return (
     <>
       <Form
         disabled={isDisabled}
         model={validationModel}
-        formValue={{
-          loginToken: sendLoginJwtExpiresMin.value,
-          passwordExpire: resetPwdJwtExpiresMin.value,
-          peeringTimeout: peeringTimeoutMs.value,
-          invoiceTries: invoiceReminderTries.value,
-          invoiceFrequency: invoiceReminderFreq.value
-        }}
-        onSubmit={async validationPassed => {
-          validationPassed && (await setShowWarning(true))
-        }}>
+        formValue={formValue}
+        onSubmit={validationPassed => validationPassed && handleSettingListUpdate()}>
```

Comment:
The setShowWarning isn't showing anymore. I assume you wanted it in the function? Also it seems like the validations aren't always preventing the form submit now because I can submit invalid data and get an error. I don't think that happened before.